tests/test_api.py

```diff
@@ -300,7 +300,13 @@ def test_empty_reporting(self) -> None:
         # empty summary reports raise exception, just like the xml report
         cov = coverage.Coverage()
         cov.erase()
-        with pytest.raises(NoDataError, match="No data to report."):
+        with pytest.raises(
+            NoDataError,
+            match=(
+                r"^The data file or directory `(.+?)` could not be found\. Perhaps `coverage "
```

Comment:
"The data file or directory" seems odd to me.  Don't we know whether we are expecting a file or directory?

Reply:
These results depend on the paths entered in the CLI. As far as I understand the code these are allowed to be data file paths or paths to directories containing data files. If the path does not exist, I'm not sure we would know if the user meant a file or a directory.